[[ Docs ]] Put language / context relevant syntax in the same API

livecodeali opened the pull request:
This patch modifies the construction of the built docs json file,
adding all entries that are livecode script syntax to the lcs
api, and similarly for livecode builder. IDE syntax is put in a
separate api since it cannot be included in a standalone.

livecodeali committed:
[[ Docs ]] Clean up dictionary functions js
Clarify state variable names
Use absolute docs indices where possible
Remember per-api previous entries
Clean up some formatting

livecodeali committed:
[[ Docs ]] Put language / context relevant syntax in the same API
This patch modifies the construction of the built docs json file,
adding all entries that are livecode script syntax to the lcs
api, and similarly for livecode builder. IDE syntax is put in a
separate api since it cannot be included in a standalone.

livecodeali committed:
[[ Docs ]] Remove API sorting
This is no longer needed now the APIs are fixed.

livecodeali commented:
Needs https://github.com/livecode/livecode/pull/6014

livecodeali committed:
[[ Docs ]] Fix display of syntax in tables
Show all syntax variants and enure there is a line break between
them.

livecodeali committed:
[[ Dictonary ]] Put different extension types in correct APIs
At the moment, we assume that a plugin extension's API should be
included in the IDE dictionary, a module extension's API should be
included in the LCB dictionary, and other extensions' APIs should
be included in the LCS dictionary.

livecodeali committed:
[[ Dictionary ]] Add release note

livecodeali committed:
[[ Dictionary ]] Don't display 'library' element

livecodeali committed:
[[ Dictionary ]] Rework entryNameToIndex function
Default links to go to the version of a piece of syntax implemented
in the same 'library'. Default all other entry lookup to the
LiveCode Script library.

livecodeali committed:
[[ Docs ]] Refactor docs generation
Also ensure all libraries have names

montegoulding commented:
Just wondering how feasible it would be to backport this to 8.2? The better search would be nice there too.

livecodeali commented:
Hmm, yeah might work - I'll have a look

livecodeali committed:
[[ Docs ]] Build distributed database correctly

livecodeali committed:
[[ Docs ]] Rework database docs API to use new 'API's

livecodeali committed:
[[ Docs ]] Lookup tokens generically
This patch enables the docs pane in the script editor to show the
docs for a token that may be contained in an extension's API

livecodeali commented:
Ok, this is rebased. Waiting on a build to see if everything works ok

livecodeali committed:
[[ Docs ]] Fix docs / alternatives rendering in SE docs pane
Now there there can be multiple entries with the same name and
type in a given API, we need to further disambiguate alternatives

livecodeali commented:
Successfully tested at https://downloads.livecode.com/livecode/staging/8.2.0.14501%20(DP%202)/g99b8a4fabb/

livecodepanos commented:
@livecode-vulcan review ok a4a4fbe

livecode-vulcan commented:
💙 review by @livecodepanos ok a4a4fbe4313b7c1746c60909601e81e730f6face
